tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py

```diff
@@ -57,9 +76,40 @@ def _insert_fused_moe_ops(gm: GraphModule) -> int:
         node.replace_all_uses_with(new_node)
         graph.erase_node(node)
 
+        _cleanup_unstacked_weights(gm, pre_fusion_param_names)
+
     return fused_key_counter
 
 
+def _cleanup_unstacked_weights(
+    gm,
+    pre_fusion_param_names,
+):
+    """DCE then unregister any now-unreferenced original params for this fusion"""
+    gm.graph.eliminate_dead_code()
+
+    used_attr_names_iter = set()
+    for n in gm.graph.nodes:
+        if getattr(n, "op", None) == "get_attr":
+            used_attr_names_iter.add(getattr(n, "target", None))
+
+    names_to_unregister_iter = [
+        name for name in pre_fusion_param_names if name not in used_attr_names_iter
+    ]
+
+    for name in names_to_unregister_iter:
+        param_obj = gm.get_parameter(name)
+        owner_mod, leaf = _resolve_owner_module_and_leaf_attr(gm, name)
+        if hasattr(owner_mod, "_parameters") and leaf in owner_mod._parameters:
+            owner_mod._parameters.pop(leaf, None)
+        if hasattr(owner_mod, leaf):
+            delattr(owner_mod, leaf)
```

Comment:
nit:
```suggestion
        delattr(owner_mod, leaf)
```
should be enough. nn.Module should clean up the rest